hydroflow_lang/src/graph/serde_graph.rs

```diff
@@ -0,0 +1,174 @@
+use slotmap::{Key, SecondaryMap, SlotMap, SparseSecondaryMap};
+
+use serde::{Deserialize, Serialize};
+
+use super::{GraphNodeId, GraphSubgraphId};
+
+#[derive(Default, Serialize, Deserialize)]
+#[allow(dead_code)] // TODO(mingwei): remove when no longer needed.
+pub struct SerdeGraph {
+    pub nodes: SecondaryMap<GraphNodeId, String>,
+    pub edges: SecondaryMap<GraphNodeId, Vec<GraphNodeId>>,
+    pub handoffs: SparseSecondaryMap<GraphNodeId, bool>,
```

Comment:
This can be `SparseSecondaryMap<GraphNodeId, ()>` to be even fancier

Reply:
I dont think edges should be Sparse as we rarely have nodes without edges.